src/data/matrix/kronecker.lean

```diff
@@ -242,10 +300,29 @@ lemma diagonal_kronecker_diagonal [mul_zero_class α]
   (diagonal a) ⊗ₖ (diagonal b) = diagonal (λ mn, (a mn.1) * (b mn.2)) :=
 kronecker_map_diagonal_diagonal _ zero_mul mul_zero _ _
 
+lemma kronecker_diagonal [mul_zero_class α] [decidable_eq n] (A : matrix l m α) (b : n → α):
+  A ⊗ₖ (diagonal b) = block_diagonal (λ i, mul_opposite.op (b i) • A) :=
+kronecker_map_diagonal_right _ mul_zero _ _
+
+lemma diagonal_kronecker [mul_zero_class α] [decidable_eq l](a : l → α) (B : matrix m n α) :
+  diagonal a ⊗ₖ B =
+    matrix.reindex (equiv.prod_comm _ _) (equiv.prod_comm _ _) (block_diagonal (λ i, a i • B)) :=
+kronecker_map_diagonal_left _ zero_mul _ _
+
 @[simp] lemma one_kronecker_one [mul_zero_one_class α] [decidable_eq m] [decidable_eq n] :
   (1 : matrix m m α) ⊗ₖ (1 : matrix n n α) = 1 :=
 kronecker_map_one_one _ zero_mul mul_zero (one_mul _)
 
+lemma kronecker_one [mul_zero_one_class α] [decidable_eq n] (A : matrix l m α) :
```

Comment:
Does it deserve to be a simp lemma?

Reply:
I think best not since it's not clear to me that `block_diagonal` is more useful to work with than `kronecker_diagonal`